packages/core/src/oidc/utils.ts

```diff
@@ -27,19 +27,21 @@ const isOrigin = (value: string) => {
   }
 };
 
+export const validateCorsAllowedOrigins = (value: unknown) => {
+  if (Array.isArray(value) && value.some((origin) => !isOrigin(origin))) {
+    throw new errors.InvalidClientMetadata(CustomClientMetadataKey.CorsAllowedOrigins);
+  }
+};
+
```

Comment:
We can use Zod for this verification?
Refer to https://github.com/logto-io/logto/pull/519/files

Reply:
If use regex to check origin, it's too complex and not readable enough.

Reply:
<img width="378" alt="image" src="https://user-images.githubusercontent.com/10594507/163095841-942b89ad-3f29-4e1a-8a29-761d9055f70c.png">

prefer to use `URL` to check origin

Reply:
Conclusion: use `zod.string().url().array()` to check

See Slack [thread](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1649821640292199).